charts/kyverno/templates/grafana/dashboard.yaml

```diff
@@ -10,4 +10,19 @@ metadata:
     {{- toYaml .Values.grafana.labels | nindent 4 }}
 data:
 {{ (.Files.Glob "grafana/*").AsConfig | indent 2 }}
+---
+{{- if .Values.grafana.grafanaDashboard.create -}}
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDashboard
+metadata:
+  name: {{ tpl .Values.grafana.configMapName . }}-dashboard
+  namespace: {{ default (include "kyverno.namespace" .) .Values.grafana.namespace }}
+spec:
+  instanceSelector:
+    matchLabels:
+      {{- toYaml .Values.grafana.grafanaDashboard.matchLabels | nindent 6 }}
+  configMapRef:
+    name: {{ tpl .Values.grafana.configMapName . }}
+    key: json
```

Comment:
I think this is the wrong value.  I believe this needs to match the key used in the config map's data.  When I run the template I get this:

```
data:
  dashboard.json: |
```

I think for this new resource you might have to put the entire resource inside a range.

```
{{- if .Values.grafana.grafanaDashboard.create -}}
{{ range $path, $_ :=  .Files.Glob  "grafana/*" -}}
<GrafanaDashboard here>
{{ end -}}
{{- end -}}
```

The unique name of the dashboard's in grafana folder would be accessible via something like this:

```
{{ $name := base $path }}
```

The idea here is that the ConfigMap technically supports more than 1 dashboard JSON so the GrafanaDashboard should too.

Reply:
yes you are totally right. thanks so much for the hint and implementation tips!
should be fixed now in https://github.com/kyverno/kyverno/pull/7992/commits/17b6e4121a1cf536ee1903af1c1e895fc49712e5